falcon/testing/client.py

```diff
@@ -2133,8 +2158,166 @@ async def __aexit__(self, exc_type, exc, tb):
         await self._task_req
 
 
+def _prepare_data_fields(data):
+    """Prepare data  fields for request body.
+
+    Args:
+        data: dict or list of tuples with json data from the request
+
+    Returns: list of 2-tuples (field-name(str), value(bytes)
+
+    """
+    fields = []
+    new_fields = []
+    if not isinstance(data, (list, dict)):
+        raise ValueError('Data must not be a list of tuples or dict.')
+    elif isinstance(data, dict):
+        fields = list(data.items())
+    else:
+        fields = list(data)
+
+    # Append data to the other multipart parts
+    for field, val in fields:
+        if isinstance(val, str) or not hasattr(val, '__iter__'):
+            val = [val]
+        for v in val:
+            if v:
+                # v has to come from json serializable obj, so can't be bytes
+                v = str(v)
+                new_fields.append(
+                    (
+                        field.decode('utf-8') if isinstance(field, bytes) else field,
+                        v.encode('utf-8') if isinstance(v, str) else v,
+                    )
+                )
+            else:
```

Comment:
how about we do like in request, and test above `if v is not None` and ignore the none case?